Samples for Extending Pentaho/Reference Implementations/Security/SAML 2.0/pentaho-saml/src/main/java/org/pentaho/platform/spring/security/saml/PentahoSamlUserDetailsService.java

```diff
@@ -206,11 +215,11 @@ public void setSelectedAuthorizationProvider( String selectedAuthorizationProvid
     this.selectedAuthorizationProvider = selectedAuthorizationProvider;
   }
 
-  public UserDetailsService getSamlUserDetailsService() {
+  public PentahoSamlNativeUserDetailsService getSamlUserDetailsService() {
     return samlUserDetailsService;
   }
 
-  public void setSamlUserDetailsService( UserDetailsService samlUserDetailsService ) {
+  public void setSamlUserDetailsService( PentahoSamlNativeUserDetailsService samlUserDetailsService ) {
```

Comment:
PentahoSamlNativeUserDetailsService implements 2 interfaces AFAICT ( SAMLUserDetailsService, UserDetailsService ).

Why are we moving from working with the contracts / interfaces to working directly with that one implementing class PentahoSamlNativeUserDetailsService ? If we are relying on some class-specific methods, then we could be breaking the abstraction layer. 

Is it possible to continue on working with the UserDetailsService interface ?